sql/core.go

```diff
@@ -341,6 +341,18 @@ type CheckAlterableTable interface {
 	DropCheck(ctx *Context, chName string) error
 }
 
+// StatementBounder allows the boundaries of an INSERT, REPLACE, UPDATE, or DELETE statement to be passed along.
+// This is modeled after the SAVEPOINT statements, but are local and specific to the aforementioned data manipulation
+// statement boundaries.
+type StatementBounder interface {
```

Comment:
StatementBounder => TableEditor

Reply:
Comment:

"TableEditor is the base interface for sub interfaces that can update rows in a table during an INSERT UPDATE or DELETE statement."